clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -797,24 +799,15 @@ public KafkaConsumer(Map<String, Object> configs,
                         config.getBoolean(ConsumerConfig.THROW_ON_FETCH_STABLE_OFFSET_UNSUPPORTED),
                         config.getString(ConsumerConfig.CLIENT_RACK_CONFIG));
             }
+            FetchConfig<K, V> fetchConfig = new FetchConfig<>(config, keyDeserializer, valueDeserializer, isolationLevel);
```

Comment:
This is a nice improvement, +1!

Reply:
```suggestion
            FetchConfig<K, V> fetchConfig = new FetchConfig<>(config, this.keyDeserializer, this.valueDeserializer, isolationLevel);
```

Reply:
@kirktrue , these should be `this.keyDeserializer` and `this.valueDeserializer`.  Otherwise the original values may be null and will cause an NPE on line 300 of `CompletedFetch.java`